src/components/form/form.js

```diff
@@ -348,6 +379,28 @@ class Form extends React.Component {
     this.activeInput = input;
   }
 
+  /**
+   * Sets the form to Dirty
+   *
+   * @method setIsDirty
+   * @return {void}
+   */
+  setIsDirty = () => {
+    this.isDirty = true;
```

Comment:
why do you need to track `this.isDirty` if you already have `this.state.isDirty`?